pyiron_gui/project/project_browser.py

```diff
@@ -23,6 +29,207 @@
 __date__ = "Feb 02, 2021"
 
 
+class PyironWrapper:
+
+    """Simple wrapper for pyiron objects which extends for basic pyiron functionality (list_nodes ...)"""
+
+    _with_self_representation = {
+        "structure": 'pyiron_atomistics.atomistics.structure.atoms.Atoms',
+        "murnaghan": 'pyiron_atomistics.atomistics.master.murnaghan.Murnaghan'
+    }
+
+    def __init__(self, pyi_obj, project, rel_path=""):
+        self._wrapped_object = pyi_obj
+        self._project = project
+        if not hasattr(pyi_obj, 'path'):
+            self._path = self._project.path
+        self._rel_path = rel_path
+        # print("init:" + self.path)
+        self._type = None
+        for key, value in self._with_self_representation.items():
+            if static_isinstance(pyi_obj, value):
+                self._type = key
+
+    @property
+    def has_self_representation(self):
+        return self._type is not None
+
+    @staticmethod
+    def _empty_list():
+        return []
+
+    @property
+    def name(self):
+        return self._type
+
+    @property
+    def project(self):
+        if hasattr(self._wrapped_object, 'project'):
+            return self._wrapped_object.project
+        return self._project
+
+    @property
+    def path(self):
+        if hasattr(self._wrapped_object, 'path'):
+            return self._wrapped_object.path
+        if hasattr(self.project, 'path'):
+            return self.project.path + '/' + self._rel_path
+        raise AttributeError
+
+    def __getitem__(self, item):
+        try:
+            return self._wrapped_object[item]
+        except (IndexError, KeyError):
+            rel_path = os.path.relpath(self.path + '/' + item, self._project.path)
+            if rel_path == '.':
+                return self._project
+            return self._project[rel_path]
+
+    def __getattr__(self, item):
+        if item in ['list_nodes', 'list_groups']:
+            try:
+                return getattr(self._wrapped_object, item)
+            except AttributeError:
+                return self._empty_list
+        return getattr(self._wrapped_object, item)
+
+    def __repr__(self):
+        return repr(self._wrapped_object)
+
+    def self_representation(self):
+        """Self representation of the wrapped object if known, else None is returned."""
+        if self._type is None:
+            return
+        if self._type == "structure":
+            return self._wrapped_object.plot3d()
+        if self._type == 'murnaghan':
```

Comment:
Nit) 103, here, and throughout: some inconsistency with ' vs "

Reply:
agreed, I don't pay enough attention to these...

Reply:
Yeah, I mix them a bunch too. It's pretty minor thing, I just like to fix it if I notice.